File: java/test_files/YesNoInput.java
import java.util.Scanner;

class YesNoInput {
    public static void main(String[] args){

        Scanner scanner = new Scanner(System.in);

        System.out.println("Do you want to continue? (yes/no)");

        String response = scanner.nextLine().toLowerCase();

        if(response.equals("yes")){
            System.out.println("You chose to continue.");
        
        }
        else if (response.equals("no")){
            System.out.println("You chose not to continue.");
        }
        else {
            System.out.println("Invalid input. Please enter 'yes' or 'no'.");
        }
    }
}

// Alternatively instead of using the built in method toLowerCase we can use the equalsIgnoreCase method
// if (response.equalsIgnoreCase("yes")){
//     System.out.println("You chose to continue.");
// } else if (response.equalsIgnoreCase("no")){
//     System.out.println("You chose not to continue.");
// } else {
//     System.out.println("Invalid input. Please enter 'yes' or 'no'.");
// }

File: CS50_c/YesNoInput.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Standardizing the input to lowercase,,,this happens char by char
void toLowerCase(char *str) {
    for (int i =0; str[i]; i++){
        str[i] = tolower(str[i]);
    }
}

int main(){
    char answer[4]; // creates a space of 4 characters to store the input
    printf("Do you want to continue? (yes/no): ");
    scanf("%3s", answer); // Limit input to 3 characters to prevent overflow
    toLowerCase(answer);

    if (strcmp(answer, "yes") == 0) {
        printf("You chose to continue.\n");
    } else if (strcmp(answer, "no") == 0) {
        printf("You chose not to continue.\n");
    } else {
        printf("Invalid input. Please enter 'yes' or 'no'.\n");
    }

    return 0;
}